workflow/scripts/parallel_vcf.py

```diff
@@ -0,0 +1,50 @@
+import os
+import argparse
+from multiprocessing import Pool
+from pysam import VariantFile
+import tempfile
+import sys
+
+parser = argparse.ArgumentParser()
+parser.add_argument("file_vcf")
+parser.add_argument("command")
+parser.add_argument("--stepsize", "-s", type=int, default=1000000)
+parser.add_argument("--threads", "-t", type=int, default=8)
+parser.add_argument("--tmp", default=None, help="Define the base directory for the tmp files")
+args = parser.parse_args()
+tmp_base = args.tmp
+stepsize = args.stepsize
+
+# create regions
+f = VariantFile(args.file_vcf)
+regions = []
+for name, contig in f.header.contigs.items():
+    last = 0
+    for x in list(range(0, contig.length, stepsize)):
+        stop = min(last + stepsize - 1, contig.length)
+        regions.append(f"{name}:{last}-{stop}")
+        last = stop+1
+f.close() 
+
+#command = "SnpSift annotate {ss_args} {args.database_vcf} /dev/stdin"
+command = args.command
+
+with tempfile.TemporaryDirectory(dir=tmp_base) as t:
+    def f(x):
+        #print("start", x, file=sys.stderr)
+        cmd = f"bcftools view -r {x} {args.file_vcf} | bcftools view -t {x} | {command} | bcftools view -Ob > {t}/{x}.bcf"
+        os.system(cmd)
+        #print("done", x, file=sys.stderr)
+        return f"{t}/{x}.bcf"
+
+    with Pool(args.threads) as p:
```

Comment:
I don't see a reason to use a threadpool here. Popen is already async, so you can simply call it several times.

Reply:
There could be thousands of parts that has to be generated, printed out as soon as they are finished (see comment below for "Why not just use bcftools concat?") and deleted. Everything in order to not screw up the sorting.
How do you manage, that only a limited number of commands are executed at the same time, while everything is print out in order without a threapool and imap? Maybe my skills for parallel execution are not developed enough.

Reply:
Publishing the script as a wrapper is problematic. Different used commands, in this case "SnpSift annotate" require different environments and with a wrapper you have it fixed. I could create a wrapper for each command, should I?
I think I will create a bioconda package.